deepchecks/base/dataset.py

```diff
@@ -124,15 +134,16 @@ def __init__(
                 if label_name in self._data.columns:
                     raise DeepchecksValueError('Can\'t pass label with label_name that exists in the data. change '
                                                'the label_name parameter')
-            self._data[label_name] = label
+            # Cast to numpy to promise no index collisions
```

Comment:
I don't like this... This is something that the user should explicitly, and make sure they understand what they're doing.

I think we should give the user an indicative error and tell them that the indices of their labels and samples don't match. It might well be that they have a problem (e.g. maybe they shuffled them differently or sampled them differently?)
Personally, I had a bug that had to do with something like that in a past DS project, and it was very annoying to find it, and it feels almost impossible to find it when this is implicitly done.

Reply:
OK, will raise a dedicated error instead if label is a series or dataframe